Detectors/EMCAL/calib/include/EMCALCalib/EMCALChannelScaleFactors.h

```diff
@@ -0,0 +1,142 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \class EMCALChannelScaleFactors
+/// \brief  Container for energy dependent scale factors for number of hits in a cell
+/// \author Joshua Koenig
+/// \ingroup EMCALCalib
+/// \since May 13, 2022
+
+#ifndef EMCAL_CHANNEL_SCALE_FACTORS_H_
+#define EMCAL_CHANNEL_SCALE_FACTORS_H_
+
+#include <cfloat>
+#include "Framework/Logger.h"
+#include "EMCALBase/Geometry.h"
+
+namespace o2
+{
+namespace emcal
+{
+
+class InvalidEnergyIntervalException final : public std::exception
+{
+
+ public:
+  /// \brief Constructor
+  /// \param E Cell energy requested
+  InvalidEnergyIntervalException(float E, int cellID) : std::exception(),
+                                                        mE(E),
+                                                        mCellID(cellID),
+                                                        mMessage("Invalid energy " + std::to_string(mE) + "for cell " + std::to_string(mCellID) + "]")
+  {
+  }
+
+  /// \brief Destructor
+  ~InvalidEnergyIntervalException() noexcept final = default;
+
+  /// \brief Access to error message
+  /// \return Error message for given exception
+  const char* what() const noexcept final { return mMessage.c_str(); }
+
+ private:
+  float mE;             ///< Cell energy requested
+  int mCellID;          ///< Cell ID
+  std::string mMessage; ///< Message to be printed
+};
+
+class EMCALChannelScaleFactors
+{
+
+ public:
+  struct EnergyIntervals {
```

Comment:
struct is usually only used for POD-objects, which are simple collections of values without additional logics in member functions. For more complicated objects one should prefer classes.